tests/AutoReview/ProjectCodeTest.php

```diff
@@ -271,12 +271,15 @@ public function testThatTestClassesAreInternal($className)
      * @param string $testClassName
      * @param string $dataProviderMethodName
      */
-    public function testThatDataProvidersAreNotPrefixedWithTest($testClassName, $dataProviderMethodName)
+    public function testThatDataProvidersAreCorrectlyNamed($testClassName, $dataProviderMethodName)
     {
-        $this->assertNotRegExp('/^test/', $dataProviderMethodName, sprintf(
-            'Data providers should not be prefixed with "test", but "%s()" in "%s" is',
+        $regExp = '/^provide[A-Z]\S+Cases$/';
+
+        $this->assertRegExp($regExp, $dataProviderMethodName, sprintf(
+            'Data provider "%s" in "%s" does not match "%s"',
             $dataProviderMethodName,
-            $testClassName
+            $testClassName,
+            $regExp
```

Comment:
can be omitted, the regex is already part of the assert fail message of PHPUnit itself, for example:
```
$this->assertRegExp('#^abc$#', 'def');
// Failed asserting that 'def' matches PCRE pattern "#^abc$#".
```